### ISSUE TITLE
Services are started by the Supervisor even after manual stop

### ISSUE THREAD

**Issue by cywang117** [balena-os/balena-supervisor#2178]
Following changes in https://github.com/balena-os/balena-supervisor/pull/2170, the Supervisor will start services which have previously been manually stopped via Supervisor API or directly through the Engine. This is a bug and an oversight made during #2170.

Steps to reproduce:
1. Push a release with restart: always:
```
version: '2.4'

services:
  always:
    restart: 'always'
    image: alpine
    command: sleep infinity
```
2. Manually stop the service using the dashboard.
3. Trigger a manual Supervisor update check using /v1/update.

Expected behavior: service should remain in `exited` status.
Actual behavior: service starts back up.

**Comment by cywang117** [comment]
After thorough discussion, we've unavoidably settled on parsing the exit message of the container, as the current implementation was too general and based on the mistaken assumption that a service status of `exited` is differentiated from a service status of `Stopped`. The former is an official service status acquirable through the container's inspect, while the latter is an internal Supervisor-only status which isn't currently set anywhere (this is another bug).

**Comment by cywang117** [comment]
Closed by #2179, available with Supervisor v14.11.8
